### ISSUE TITLE
Socket select errset non empty

### ISSUE THREAD

**Issue by ftx** [Azure-Samples/iot-hub-c-esp32#3]
Hello,

I have make firmware with your example and, when i launch the device, i have a problem.
He connect to the wifi and iot hub, and error...

Do you have any ideas of this error?
Thanks

`Info: Initializing SNTP

Info: SNTP initialization complete

Info: IoT Hub SDK for C, version 1.1.22

IoTHubClient_LL_SetMessageCallback...successful.

IoTHubClient_LL_SendEventAsync accepted message [0] for transmission to IoT Hub.

IoTHubClient_LL_SendEventAsync accepted message [1] for transmission to IoT Hub.

I (13777) wifi: pm start, type:0

I (13777) wifi: active cnt: 20
I (23777) wifi: active cnt: 3
Error: Time:Thu Sep  7 20:34:40 2017 File:/home/flotix/esp/esp-idf/components/azure-iot/sdk/c-utility/adapters/socket_async.c Func:socket_async_is_create_complete Line:196 Socket select errset non-empty: 113

Info: socket_async_is_create_complete failure

Guru Meditation Error of type InstrFetchProhibited occurred on core  1. Exception was unhandled.
Register dump:
PC      : 0x00000000  PS      : 0x00060c30  A0      : 0x8011fdc4  A1      : 0x3ffc8080  
A2      : 0x3ffbc00c  A3      : 0x3f410cd4  A4      : 0x3f411230  A5      : 0x000002b8  
A6      : 0x00000001  A7      : 0x3f410ddc  A8      : 0x8011fbe5  A9      : 0x3ffc8060  
A10     : 0x00000000  A11     : 0x00000001  A12     : 0x00000000  A13     : 0x00000020  
A14     : 0x00000001  A15     : 0x3f41300c  SAR     : 0x0000000c  EXCCAUSE: 0x00000014  
EXCVADDR: 0x00000000  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0xffffffff  

Backtrace: 0x00000000:0x3ffc8080 0x4011fdc1:0x3ffc80a0 0x4011ffa2:0x3ffc80d0 0x40144153:0x3ffc80f0 0x4011eb48:0x3ffc8110 0x4011d1dd:0x3ffc8140 0x4011ab35:0x3ffc8170 0x401188a7:0x3ffc8190 0x40101347:0x3ffc81c0 0x40100ec4:0x3ffc8230

Rebooting...
ets Jun  8 2016 00:22:57
`

**Comment by JetstreamRoySprowl** [comment]
I just ran the sample with the latest Azure IoT and Espressif SDKs and am not seeing this problem. When I built I used all of the menuconfig defaults except for COM port, SSID, and password. The combination of the surprising socket select failure plus the subsequent reboot is the sort of thing one might expect from running out of memory. Is there other code in the device besides the Azure IoT sample?
